akka-http-core/src/main/scala/akka/http/scaladsl/model/MediaType.scala

```diff
@@ -136,10 +136,10 @@ object MediaType {
   def customWithFixedCharset(mainType: String, subType: String, charset: HttpCharset, fileExtensions: List[String] = Nil,
                              params:                 Map[String, String] = Map.empty,
                              allowArbitrarySubtypes: Boolean             = false): WithFixedCharset = {
-    require(mainType != "multipart", "Cannot create a MediaType.Multipart here, use `customMultipart` instead!")
-    require(allowArbitrarySubtypes || subType != "*", "Cannot create a MediaRange here, use `MediaRange.custom` instead!")
-    val _params = params
-    new WithFixedCharset(renderValue(mainType, subType, params), mainType, subType, charset, fileExtensions) {
+    val (_params, _mainType, _subType) = (params, mainType.toRootLowerCase, subType.toRootLowerCase)
+    require(_mainType != "multipart", "Cannot create a MediaType.Multipart here, use `customMultipart` instead!")
+    require(allowArbitrarySubtypes || _subType != "*", "Cannot create a MediaRange here, use `MediaRange.custom` instead!")
+    new WithFixedCharset(renderValue(_mainType, _subType, params), _mainType, _subType, charset, fileExtensions) {
       override def params = _params
       override def isApplication = mainType == "application"
```

Comment:
Should these be updated to refer to `_mainType` as well?

Reply:
Here `mainType` references class field since we have it inside `new WithFixedCharset(... val mainType: String, val subType: String ... )`, the value is passed with `_mainType` as a constructor parameter.

Reply:
Ah, you're right of course!